internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -773,35 +775,66 @@ func expandLabelSliceFromSetSchema(d *schema.ResourceData) ([]admin.ComponentLab
 	return res, nil
 }
 
-func expandAdvancedReplicationSpecs(tfList []any) *[]admin.ReplicationSpec {
+func expandAdvancedReplicationSpecs(tfList []any, rootDiskSizeGB *float64) *[]admin.ReplicationSpec20240710 {
 	if len(tfList) == 0 {
 		return nil
 	}
-	var apiObjects []admin.ReplicationSpec
+	var apiObjects []admin.ReplicationSpec20240710
 	for _, tfMapRaw := range tfList {
 		tfMap, ok := tfMapRaw.(map[string]any)
 		if !ok || tfMap == nil {
 			continue
 		}
-		apiObject := expandAdvancedReplicationSpec(tfMap)
+		apiObject := expandAdvancedReplicationSpec(tfMap, rootDiskSizeGB)
+		apiObjects = append(apiObjects, *apiObject)
+
+		// handles adding additional replication spec objects if legacy num_shards attribute is being used and greater than 1
+		numShards := tfMap["num_shards"].(int)
```

Comment:
Would be nice to see some test examples of how the added replication_specs looks when `num_shards > 1` ,I think the replication IDs might need to be added for the update calls, but that probably goes in the next "update" PR?

Reply:
as mentioned [here](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2363#discussion_r1654889746), we will need to adjust the read operation to consume the new API, and the support the update as well. As these operations are implemented we can start including tests capturing the new scenarios.